docs/2023/README.md

```diff
@@ -0,0 +1,536 @@
+
+### Keyset Proof
+
+The `keyset_proof` is the proof that the device has authority to participate in some keyset. The keyset defines the rules which determine the methods for revoking or replacing its keys.
+
+The first entry the app makes in each user's source chain is a `KeysetRoot`, creating a new keyset space.
+
+A source chain may later reference a valid `DeviceInvite`, in the form of a `DeviceInviteAcceptance`, to abandon the initial keyset and join another already existing keyset space.
+
+(This will be at least the fifth entry in the chain, after the three genesis entries and the `init_complete`.)
+
+If the keyset proof is a new `KeysetRoot` then it must be immediately followed by a valid `ChangeRule` to define how key management works within this keyset. On the other hand, if you join an existing keyset through a `DeviceInvite`, the `ChangeRule` of that keyset is what governs keys made on this chain.
+
+## Keyset
+
+A keyset is the set of keys governed by a ruleset, presumably under the control of one person.
+
+When you install a new app in Holochain, by default a new keypair is generated to control the source chain of that app. The public key of that keypair serves as the address of your agent in that app's DHT. The private key signs all of your network communications and all actions on your chain. Deepkey registers, manages, and reports on the validity of each `AgentPubKey` installed on your conductor.
+
+### Keyset Root
+
+A `KeysetRoot` (KSR) is self-declared onto the network using a single-purpose throwaway keypair.
+
+The structure of a `KeysetRoot` is:
+
+- The `first_deepkey_agent` (FDA), the author of the `KeysetRoot`.
+- The `root_pub_key`, the public part of a throwaway keypair which is only used to generate this KSR. (Using the `sign_ephemeral` HDK function.)
+- A `Signature`: the authority of the FDA is established using the private part of the throwaway keypair to sign the FDA's pubkey.
+
+Note that if a device is to issue a KSR it must do so as its very first action in Deepkey.
+
+**A device can NEVER create another KSR without starting a new source chain.**
+
+#### KeysetRoot API
+
+**Create**: The validation that happens when you create a new `KeysetRoot`
+
+- A `KeysetRoot` struct must deserialize cleanly from the record being validated.
+- Must be created at index 4 (5th item) in the author's chain.
+- The author must be the FDA.
+- The signature of the FDA from the root/ephemeral pubkey must be valid.
+
+**Read**: There is currently no read functions or lookups exposed as zome calls, but this may change in the future as people may want to use `KeysetRoot` as a unifying identity.
+
+**Update**: Not allowed.
+
+**Delete**: Not allowed.
+
+**Zome Calls**:
+
+- `create_keyset_root`
+  - Input is a `(KeysetRoot, ChangeRule)` tuple.
+  - Creates both the `KeysetRoot` and `ChangeRule` records sequentially.
+  - Output is a `(ActionHash, ActionHash)` tuple of the created records.
```

Comment:
API input is (or will be) captured in 2 places

- rust cargo docs that will be published on Github pages
- javascript examples that will be in the `./docs` directory